src/main/resources/templates/search.html

```diff
@@ -0,0 +1,77 @@
+<!DOCTYPE html>
+<html xmlns:th="http://www.thymeleaf.org/">
+<head th:replace="fragments :: head">
+</head>
+<body>
+
+<div class="container body-content">
+
+    <h1 th:text="${#strings.capitalizeWords(title)}"></h1>
+
+    <form th:action="@{/search/results}" method = "post">
+        <h2>Search by:</h2>
+
+
+        <p>
+            <span th:each="column : ${columns}">
+                <input
+                        type="radio"
+                        name="searchType"
+                        th:id="${column.key}"
+                        th:value="${column.key}"
+                        th:checked="${column.key == 'all'}"/>
+                <label th:for="${column.key}" th:text="${column.value}"></label>
+            </span>
+
+        </p>
+
+        <p>
+            <label for="searchTerm">KeyWord:</label>
+            <input type="text" name="searchTerm" id="searchTerm" />
+
+        </p>
+
+        <input type="submit" value="Search" />
+        <a href=" /adv-search">Advanced Search</a>
+
+    </form>
+
+    <hr />
+
+
+    <table class="recipe-listing">
+        <tr th:each="recipe : ${recipes}">
```

Comment:
With the way this table is set up, there are no table header <th> elements, each of the table rows <tr> only holds one piece of table data <td> which is a series of <p> tags. This may be better displayed using other methods, such as a loop of <div> elements instead.